src/main/resources/META-INF/rewrite/java.version-11.yml

```diff
@@ -0,0 +1,25 @@
+#
+# Copyright 2022 the original author or authors.
+# <p>
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# <p>
+# https://www.apache.org/licenses/LICENSE-2.0
+# <p>
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+---
+type: specs.openrewrite.org/v1beta/recipe
+name: org.openrewrite.java.migrate.JavaVersion11
+displayName: Change Maven property java.version to 11
+description: Change the Maven compiler source and target versions to 11, via Spring Boot property java.version.
```

Comment:
Was doubting to fully remove any mention of Spring here, as I'm not sure if the property is used as a convention in any other framework. So without the mention of Spring, we would also have to remove the maven compile source & target reference, as there's not guarantee it does anything at all if Spring is not present.

Reply:
Either way, welcome to change this as you see fit; I'll be asleep for the next few hours :)

Reply:
🤔 Yeah, you are right, I guess this property is more a "spring" thing.

I think it might be safer to add all three property changes (and only make the change if the property already exists) 

- `java.version`
- `maven.compiler.source`
- `maven.compiler.target`

That will take care of the three most common use cases. 

Then beyond this, Once we have added plugin/pluginManagement to the semantic model for maven, this can be converted to an imperative recipe, so we can make decisions like "if the property is not defined, add it to the top-level pom"

Reply:
Pushed a change to now also change the values of the maven compiler properties if present.

I'd be inclined to rename the recipe, displayName & description to reflect that we are changing the maven compiler source & target version, rather than any Spring/Java specific property. That's essentially what we aim to do here, with the Spring specifics just being an implementation detail for some projects.

Welcome to weigh in or make any changes to wrap this up. :)